cf/client.go

```diff
@@ -196,3 +198,26 @@ func (l *logCacheHTTPClient) Do(req *http.Request) (*http.Response, error) {
 
 	return l.client.Do(req)
 }
+
+func getTokenWithRetry(tokenSource oauth2.TokenSource, maxRetries int, fallOffSeconds time.Duration) (*oauth2.Token, error) {
+	var (
+		i     int
+		token *oauth2.Token
+		err   error
+	)
+
+	for i = 0; i < maxRetries; i++ {
+		token, err = tokenSource.Token()
+
+		if err == nil {
+			return token, err
+		}
+
+		log.Println(fmt.Sprintf("getting token failed (attempt %d of %d). Retrying. Error: %s", i, maxRetries, err.Error()))
```

Comment:
Why not `log.Printf`? Why `log.Println` and `fmt.Sprintf`?

Reply:
Good question. Don't think I was looking too much when I wrote this!

Reply:
Addressed